src/common/fork_function.h

```diff
@@ -0,0 +1,152 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+// Run a function in a forked child, with a timeout.
+
+#pragma once
+
+#include <functional>
+#include <signal.h>
+#include <sys/wait.h>
+#include <sys/types.h>
+#include <common/errno.h>
+#include <ostream>
+#include "common/errno.h"
+
+static void _fork_function_dummy_sighandler(int sig) {}
+
+// Run a function post-fork, with a timeout.  The exit code encoding
+// weirdness does seem to want me to allow negative values, so we make
+// the std::function unsigned.  fork_function() itself is signed
+// because it may return ETIMEDOUT.
+
+static inline int fork_function(
+  int timeout,
+  std::ostream& errstr,
+  std::function<unsigned(void)> f)
+{
+  // first fork the forker.
+  pid_t forker_pid = fork();
+  if (forker_pid) {
+    // just wait
+    int status;
+    while (waitpid(forker_pid, &status, 0) == -1) {
+      assert(errno == EINTR);
+    }
+    if (WIFSIGNALED(status)) {
+      errstr << ": got signal: " << WTERMSIG(status) << "\n";
+      return 128 + WTERMSIG(status);
+    }
+    if (WIFEXITED(status)) {
+      int r = WEXITSTATUS(status);
```

Comment:
@liewegas 
WEXITSTATUS(status) evaluates to the low-order 8 bits of the argument passed to exit() by the child.
We could make the std::function `int8_t`, and change this to:
```
    int8_t r = WEXITSTATUS(status);
```
I suppose this would allow to avoid "unsigned awkwardness".